src/bin/wd_copier.rs

```diff
@@ -12,10 +15,67 @@ struct Opt {
     dst: PathBuf,
 }
 
+/// Get OS dirty byte count using procfs::Meminfo
+fn get_dirty_bytes() -> u64 {
+    match procfs::Meminfo::new() {
+        Ok(o) => o.dirty,
+        Err(_e) => 0,
+    }
+}
+
+/// Calculates the percentage of max as it approaches min
+///
+/// # Arguments
+///
+/// * `current` - current value of progress
+/// * `min` - target; percent = 100 when current <= min
+/// * `max` - starting point; percent = 0 when current == max
+fn calc_percent(current: u64, min: u64, max: u64) -> u64 {
```

Comment:
You could consider using a https://doc.rust-lang.org/std/ops/struct.RangeInclusive.html for min and max.

Reply:
Is there an actual advantage to using this over `u64`? Mostly asking since I'm not using an iterator in this function, you end up just changing it to

```
fn calc_percent(current: u64, range: RangeInclusive<u64>) -> u64 {
     // Subtract min from current but clamp to 0u64
     let numerator = current.saturating_sub(range.start());                                                                                      
     let denominator = range.end() / 100;
```
Unless there's something I'm missing

Reply:
Yep, the advantage just to make it hard to pass parameters in the wrong order. I like to avoid having multiple parameters of the same type next to each other, for example you could easily pass in `min, max, current` and not notice that it's supposed to be `current, min, max`. By using a range for the min/max, you make it obvious at the call site that the order is correct.